torch/jit/_monkeytype_config.py

```diff
@@ -64,22 +64,22 @@ def consolidate_types(self, qualified_name: str) -> Dict:
             # then consolidate the type to `Any` and replace the entry
             # by type `Any`.
             for arg, types in all_args.items():
-                _all_type = " "
-                for _type in types:
+                all_type = " "
+                for arg_type in types:
                     # If the type is a type imported from typing
                     # like Tuple, List, Dict then replace "typing."
                     # with a null string.
-                    if inspect.getmodule(_type) == typing:
-                        _type_to_string = str(_type)
-                        _all_type += _type_to_string.replace('typing.', '') + ','
+                    if inspect.getmodule(arg_type) == typing:
+                        type_to_string = str(arg_type)
+                        all_type += type_to_string.replace('typing.', '') + ','
                     else:
-                        _all_type += _type.__name__ + ','
-                _all_type = _all_type.lstrip(" ")  # Remove any trailing spaces
+                        all_type += get_qualified_name(arg_type) + ','
```

Comment:
Using a `List[str]` instead of `set` results in duplicate values being added to the  `List` which needs to be handled. Maybe instead of adding the arg_type ,  I can add the qualified name of the type in `analyze`.

Reply:
I don't quite understand what you mean.

`all_type` is currently a string, de-dup is done before when constructing `all_args`.